prow/apis/prowjobs/v1/types.go

```diff
@@ -163,6 +163,16 @@ type ProwJobSpec struct {
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
 }
 
+type TektonSpec struct {
+	// GitResourceName is the name to use when binding the prow git resource
+	GitResourceName string
```

Comment:
Is the main goal of this change to ruse the same pipeline definition across different prow jobs? Do I understand correctly?

At the moment, the ProwJob and the PipelineRun do not get deleted immediately after the build is completed. This means the PipelineRun name needs to be unique per prow job execution. This is the reason for using the prow job name as the pipeline run name. It is also easier to debug.

https://github.com/kubernetes/test-infra/blob/19105951765d4456a7c258c3f96312ecd0da0bed/prow/cmd/pipeline/controller.go#L528

Also a new GitResource gets created for every prow job using the actual git reference from the prow job.

https://github.com/kubernetes/test-infra/blob/19105951765d4456a7c258c3f96312ecd0da0bed/prow/cmd/pipeline/controller.go#L565

For instance, when a new git commit in pushed into the PR the git reference is updated in the prow job. 

I think in case you want to use a fixed name defined in the config for GitResource, most likely you need also to update the `revision` in the GitResource with every job execution otherwise it will point to the first git reference which created the job.

Reply:
> Is the main goal of this change to ruse the same pipeline definition across different prow jobs? Do I understand correctly?

It's not a *ruse* though. `Pipeline` and `Task` are meant to be re-used (with different parameters and resources). The goal is to make sure we can use a `Pipeline` that defines a `foo` git pipelineresource with `prow`. The PR adds only that, a.k.a. letting the user use a pre-defined name for the binding of the gitresource (not the name of the gitresource itself)

It's ok to create a PipelineResource for each prowjobs for now.
- we could be smarter and look if a pipelineresource for this particular details already exists — but it's just a nice to have.
- we could/have to think of a garbage collection for the pipelineresource

> At the moment, the ProwJob and the PipelineRun do not get deleted immediately after the build is completed. This means the PipelineRun name needs to be unique per prow job execution. This is the reason for using the prow job name as the pipeline run name. It is also easier to debug.

Yeah this is perfect. The `PipelineRun` name has to be unique, the `Pipeline` on the other hand (what we refer in `PipelineRunSpec.PipelineRef`) do not need to be unique. `Pipeline` holds the definition of the pipeline, that can be *instantiate* any number of time by `PipelineRun`s.

The flow is : 
- You have re-usable `Task`s (definition) in your cluster, that refer, in our case, to a git resource (that needs to be binded at runtime).
- You have your *re-useable* `Pipeline` (definition) in you cluster, that refer, in our case, to a git resource (that needs to be binded at runtime — that will be passed to the tasks). The git resource the pipeline refers, has to have a name, that is stable, and that the `PipelineRun` will use to bind a `PipelineResource` to it.
- For each run (her ProwJobs), we create (or re-use) a `PipelineResource` (of type git), and a `PipelineRun` that binds this `PipelineResource` to the required resource the `Pipeline` needs.


Next step will to be able to apply them (`Task`s and `Pipeline`s) from the repository itself but this is not the goal for now (small steps :D)

Reply:
> The PR adds only that, a.k.a. letting the user use a pre-defined name for the binding of the gitresource (not the name of the gitresource itself)

Thanks for clarification. I get it now :). It's just the `binding` name. 

> we could/have to think of a garbage collection for the pipelineresource

Probably, we can extend sinker to remove the pipeline resources when a `ProwJob` is deleted. Another option which was discussed (and it was initially implemented) is to make the `ProwJob` as owner of all these resources and then rely on k8s garbage collection. It doesn't seem to work when using a different build cluster for Tekton's resources.

> Next step will to be able to apply them (Tasks and Pipelines) from the repository itself but this is not the goal for now (small steps :D)

This would be nice! In Jenkins-X, the pipeline definition is stored into the repository and it gets transpiled into Tekton CRDs on the fly. The same can be done with the raw Tekton CRDs.